slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/validator/AttestationValidator.java

```diff
@@ -0,0 +1,147 @@
+    this.signingRoot = signingRoot;
+    this.sourceEpoch = sourceEpoch;
+    this.targetEpoch = targetEpoch;
+    this.validatorId = validatorId;
+    this.signedAttestationsDao = signedAttestationsDao;
+  }
+
+  public boolean sourceGreaterThanTargetEpoch() {
+    if (sourceEpoch.compareTo(targetEpoch) > 0) {
+      LOG.warn(
+          "Detected sourceEpoch {} greater than targetEpoch {} for {}",
+          sourceEpoch,
+          targetEpoch,
+          publicKey);
+      return false;
+    }
+    return true;
+  }
+
+  public boolean existsInDatabase() {
+    return signedAttestationsDao
+        .findMatchingAttestation(handle, validatorId, targetEpoch, signingRoot)
+        .isPresent();
+  }
+
+  public void insertToDatabase() {
+    final SignedAttestation signedAttestation =
+        new SignedAttestation(validatorId, sourceEpoch, targetEpoch, signingRoot);
+    signedAttestationsDao.insertAttestation(handle, signedAttestation);
+  }
+
+  public boolean directlyConflictsWithExistingEntry() {
+    return !signedAttestationsDao
+        .findAttestationsForEpochWithDifferentSigningRoot(
+            handle, validatorId, targetEpoch, signingRoot)
+        .isEmpty();
+  }
+
+  public boolean hasSourceOlderThanWatermark() {
+    final Optional<UInt64> minimumSourceEpoch =
+        signedAttestationsDao.minimumSourceEpoch(handle, validatorId);
+    if (minimumSourceEpoch.map(minEpoch -> sourceEpoch.compareTo(minEpoch) < 0).orElse(false)) {
+      LOG.warn(
+          "Attestation source epoch {} is below minimum existing attestation source epoch {}",
+          sourceEpoch,
+          minimumSourceEpoch.get());
+      return true;
+    }
+    return false;
+  }
+
+  public boolean hasTargetOlderThanWatermark() {
+    final Optional<UInt64> minimumTargetEpoch =
+        signedAttestationsDao.minimumTargetEpoch(handle, validatorId);
+    if (minimumTargetEpoch.map(minEpoch -> targetEpoch.compareTo(minEpoch) <= 0).orElse(false)) {
+      LOG.warn(
+          "Attestation target epoch {} is below minimum existing attestation target epoch {}",
+          targetEpoch,
+          minimumTargetEpoch.get());
+      return true;
+    }
+    return false;
+  }
+
+  public boolean surroundsExistingAttestation() {
+    // check that no previous vote is surrounded by attestation
+    final Optional<SignedAttestation> surroundedAttestation =
+        signedAttestationsDao.findSurroundedAttestation(
+            handle, validatorId, sourceEpoch, targetEpoch);
+    if (surroundedAttestation.isPresent()) {
+      LOG.warn(
+          "Detected surrounded attestation {} for attestation signingRoot={} sourceEpoch={} targetEpoch={} publicKey={}",
+          surroundedAttestation.get(),
+          signingRoot,
+          sourceEpoch,
+          targetEpoch,
+          publicKey);
+      return true;
+    }
+    return false;
+  }
+
+  public boolean isSurroundedByExistingAttestation() {
+    // check that no previous vote is surrounding the attestation
```

Comment:
nit: this comment isn't needed now this is split out into it's own function

Reply:
done